### PR TITLE
Update to operator-sdk v0.18.1, and gen CRDs

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Switches the CRDs to the v1 API

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

```release-note
Update CRDs to v1
```

### CODE REVIEW

File: deploy/crds/operator.tigera.io_amazoncloudintegrations_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (caseydavenport) [inline deploy/crds/operator.tigera.io_amazoncloudintegrations_crd.yaml:1]: It's unfortunate that it uses a different name for the CRD. Let me see if I can trick git into treating this as a "moved" file.

File: pkg/apis/operator/v1/intrusiondetection_types.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ type IntrusionDetectionStatus struct {
 // of this resource is supported. It must be named "tigera-secure".
 // +k8s:openapi-gen=true
 // +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
```
Comment (tmjd) [inline pkg/apis/operator/v1/intrusiondetection_types.go:45]: Is the `+genclient:nonNamespaced` still needed?

File: Makefile [hunk 3]
```diff
@@ -417,10 +417,9 @@ endif
 ###############################################################################
 # Utilities
 ###############################################################################
-## Generating code after API changes
+## Generating code after API changes. Requires operator-sdk v0.18.1.
 gen-files:
-	operator-sdk generate k8s
-	operator-sdk generate openapi
```
[the remark this answers is not in the document]
Comment (caseydavenport) [reply Makefile:423]: This command no longer exists in the latest operator-sdk. I think that this is the command that actually used to generate crds (there wasn't a generate crds subcommand before)

File: pkg/apis/operator/v1/intrusiondetection_types.go (hunk 2 above)
Comment (caseydavenport) [reply pkg/apis/operator/v1/intrusiondetection_types.go:45]: The `genclient` stuff is all for building k8s golang clients.

The `kubebuilder` tags are for CRD yaml files. 

So, I think they are both used for different things.

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (tmjd) [reply Makefile:423]: There is still a k8s generate that is needed to `Generates Kubernetes code for custom resource`, without that we won't see our deepcopy code generated.

**Pull request merged**